mountpoint-s3/src/upload.rs

```diff
@@ -136,9 +149,25 @@ impl<Client: ObjectClient> UploadRequest<Client> {
     pub async fn complete(self) -> Result<PutObjectResult, PutRequestError<Client>> {
         let size = self.size();
         let checksum = self.hasher.finalize();
-        self.request
+        let result = self
+            .request
             .review_and_complete(move |review| verify_checksums(review, size, checksum))
-            .await
+            .await?;
+        self.sse
+            .verify_response(result.sse_type.as_deref(), result.sse_kms_key_id.as_deref())
+            .unwrap_or_else(|err| {
```

Comment:
closure with side effects also looks like a bad pattern to me, at least in case of `unwrap_or_else` it makes sense that a closure should solely process unwrapped values